src/main/java/org/gradle/profiler/GradleScenarioInvoker.java

```diff
@@ -69,7 +67,14 @@ public void doRun(GradleScenarioDefinition scenario, InvocationSettings settings
         try {
             buildConfiguration.printVersionInfo();
 
-            List<String> allBuildsJvmArgs = new ArrayList<>(buildConfiguration.getJvmArguments());
+            List<String> allBuildsJvmArgs = new ArrayList<>();
+            if (!scenario.getJvmArgs().isEmpty()) {
+                allBuildsJvmArgs.addAll(buildConfiguration.getJvmArguments().stream().filter(arg -> !arg.startsWith("-X")).collect(Collectors.toList()));
```

Comment:
❌ There should be no filtering logic here.

Reply:
<img width="418" alt="Screen Shot 2019-04-11 at 7 10 54 PM" src="https://user-images.githubusercontent.com/1349353/56003779-cebabf00-5c8d-11e9-8918-7f85c9f3e2e8.png">

It looks like we will end up with two -Xmx (and possibly other) args if we don't filter, although the test still pass. Does the last argument always win?

Reply:
Yes that's how the jvm and most other command line programs handle arguments